File: src/com/moneydance/modules/features/yahooqt/BasePropertyChangeReporter.java
package com.moneydance.modules.features.yahooqt;

import javax.swing.event.SwingPropertyChangeSupport;
import java.beans.PropertyChangeListener;

public class BasePropertyChangeReporter {
    public static final String ALL_PROPERTIES = "UpdateAll";
    protected final SwingPropertyChangeSupport eventNotify = new SwingPropertyChangeSupport(this);

    public BasePropertyChangeReporter() {
    }

    public void addPropertyChangeListener(PropertyChangeListener listener) {
        this.eventNotify.addPropertyChangeListener(listener);
    }

    public void removePropertyChangeListener(PropertyChangeListener listener) {
        this.eventNotify.removePropertyChangeListener(listener);
    }

    protected void notifyPropertyChanged(String propertyName, Object oldValue, Object newValue) {
        this.eventNotify.firePropertyChange(propertyName, oldValue, newValue);
    }

    public void notifyAllListeners() {
        this.eventNotify.firePropertyChange("UpdateAll", (Object)null, (Object)null);
    }
}

File: src/com/moneydance/modules/features/yahooqt/BasePropertyChangeReporter.kt
package com.moneydance.modules.features.yahooqt

import java.beans.PropertyChangeListener
import javax.swing.event.SwingPropertyChangeSupport

open class BasePropertyChangeReporter {
  protected val eventNotify: SwingPropertyChangeSupport = SwingPropertyChangeSupport(this)
  
  fun addPropertyChangeListener(listener: PropertyChangeListener?) {
    eventNotify.addPropertyChangeListener(listener)
  }
  
  fun removePropertyChangeListener(listener: PropertyChangeListener?) {
    eventNotify.removePropertyChangeListener(listener)
  }
  
  protected fun notifyPropertyChanged(propertyName: String?, oldValue: Any?, newValue: Any?) {
    eventNotify.firePropertyChange(propertyName, oldValue, newValue)
  }
  
  fun notifyAllListeners() {
    eventNotify.firePropertyChange("UpdateAll", null as Any?, null as Any?)
  }
  
  companion object {
    const val ALL_PROPERTIES: String = "UpdateAll"
  }
}
